Assets/MixedRealityToolkit.SDK/CompilationValidator/CompilationValidator.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using System;
+using System.Diagnostics;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    [InitializeOnLoad]
+    public class CompilationValidator
+    {
+        private const string warningTitle = "Assembly reference missing";
+        private const string warningMessage = "Unity 2019.2 and later requires that your UnityAR assembly definition be configured differently. We can attempt to do this automatically. Proceed?";
+        private const string unityArAsmdefPath = "Assets/MixedRealityToolkit.Staging/UnityAR/Microsoft.MixedReality.Toolkit.Providers.UnityAR.asmdef";
+        private const string spatialTrackingReference = "UnityEngine.SpatialTracking";
+        private const string powershellScriptPath = "scripts/support2019/setup_for_2019.ps1";
+
+        static CompilationValidator()
+        {
+#if UNITY_2019_2_OR_NEWER
+            TextAsset unityArAsmdef = AssetDatabase.LoadAssetAtPath<TextAsset>(unityArAsmdefPath);
```

Comment:
This seems like something that could maybe fit in with MixedRealityEditorSettings.cs?

Reply:
That way we can also get rid of the asmdef. Speed up the VS solution open a bit

Btw, out of curiosity why did you put this in the SDK package instead of core?

Reply:
Actually scratch above. If this is github repo only, merging with MREditorSettings might not be a good idea. 

@davidkline-ms is there a way to signify that this file should not be built/included with the unitypackages process or nuget etc?